ibis/common/annotations.py

```diff
@@ -288,9 +288,8 @@ def from_callable(cls, fn, validators=None, return_validator=None):
 
         if return_validator is not None:
             return_annotation = return_validator
-        elif sig.return_annotation is not EMPTY:
-            typehint = evaluate_typehint(sig.return_annotation, fn.__module__)
-            return_annotation = Validator.from_typehint(typehint)
+        elif "return" in typehints:
+            return_annotation = Validator.from_typehint(typehints["return"])
```

Comment:
```suggestion
        elif (typehint := typehints.get("return")) is not None:
            return_annotation = Validator.from_typehint(typehint)
```

Maybe this doesn't matter in the grand scheme of things, but you can avoid the extra lookup here for about the same amount of code 🤷🏻